vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,556 @@
+
+-   Here is the code -
+
+    ```         
+    box::use(
+      ggplot2[ggplot, aes, geom_bar, labs, theme_light],
+    )
+
+    #' @export
+    generate_barchart <- function(filtered_data) {
+            ggplot(filtered_data, aes(x = rep_name, y = deal_amount, fill = rep_name)) +
+            geom_bar(stat = "identity") +
+            labs(x = "Sales Rep Name", y = "Deal Amount") +
+            theme_light()
+    }
+    ```
+
+-   The above code block loads the **`ggplot2`** package and imports several functions from it.
+    It then defines a function called **`generate_barchart()`** that takes a **`filtered_data`** argument as input.
+    The function uses the **`ggplot()`** function to create a bar chart with **`x = rep_name`** and **`y = deal_amount`**, where **`rep_name`** and **`deal_amount`** are columns in the **`filtered_data`** data frame.
+    The **`fill`** aesthetic is also set to **`rep_name`**, which maps each bar to a unique color based on the **`rep_name`** variable.
+    The function also adds labels to the x and y axes and applies the **`theme_light()`** theme to the plot.
+    Finally, the function is exported with **`#' @export`**, which makes it available to other packages and scripts that import this one.
+
+### Call the helper function in the desired `rhino` module
+
+-   Now let us go back to the `app\view\datatable.R` file.
+    Here we need to call the `generate_barchart()` function we just created along with the required dependencies to render it on UI as a plotly bar chart.
+
+-   So, the code begins with
+
+    ```         
+    box::use(
+      shiny[moduleServer, NS, uiOutput, renderUI, reactive, div],
+      shiny.fluent[fluentPage, Text, DetailsList, Toggle.shinyInput],
+      tibble[tibble],
+      dplyr[filter],
+      plotly[plotlyOutput, renderPlotly],
+    )
+
+    box::use(
+      app/logic/barchart[generate_barchart],
+    )
+    ```
+
+    Here, we have called the `plotly` library, and also the function that we defined in the `app/logic` folder.
+
+-   Now, we will add one more `div` in the ui section after the datatable to make room for the bar chart
+
+````         
+```         
+ div(
+      class = "ms-depth-64",
+      style = "height: 500px; overflow: auto;",
+      plotlyOutput(ns("barchart"))
+      )
+```
+
+`plotlyOutput` will help us to render the graph
+````
+
+-   Similarly, in the server section, we will need to add some code to ask the backend of the application for rendering the barchart
+
+    ```         
+     output$barchart <- renderPlotly({
+            generate_barchart(filtered_deals())
+        })
+    ```
+
+Once ready, on running the application, we should see the plotly bar chart rendering successfully and also responding to our interactions with the toggle input button.
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_4.png")
+```
+
+## 6. Some Improvements for better User Experience
+
+### Keep the data table and bar chart on the same row
```

Comment:
Why you are not using Fluent components like Card as containers for the application elements?

Reply:
with reference to the this [tutorial](https://youtu.be/KKWAeeZM4Rc?t=1793), card looks like a function created to build containers. Hence directly demonstrated with divs inside code.

Reply:
And adding a custom component based on Fluent is a perfect fit for the tutorial, so you can reuse it.